homeassistant/components/risco/__init__.py

```diff
@@ -154,6 +154,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
+        if is_local(entry):
+            local_data: LocalData = hass.data[DOMAIN][entry.entry_id]
+            await local_data.system.disconnect()
```

Comment:
Does this ever happen in the tests?  If so it needs to be patched out

Reply:
I checked test runtime before committing, so it looks like `disconnect()` is patched whenever this is used. It does make sense to add a specific test to make sure it's actually called, though

Reply:
added